lib/cell.rb

```diff
@@ -28,19 +28,36 @@ def fire_upon
     @fired_upon = true
   end
 
-  def render
-    if @ship != nil
-      if @ship.sunk?
-        "X"
-      else 
-        "H"
-      end
+  # def render
+  #   if @ship != nil
+  #     if @ship.sunk?
+  #       "X"
+  #     else 
+  #       "H"
+  #     end
+  #   else
+  #     if self.fired_upon?
+  #       "M"
+  #     else
+  #       "."
+  #     end
+  #   end
+  # end
+
+  # def fire_upon
+  #   @fired_upon = true
+  #   @ship.hit unless empty?
+  # end
+
+  def render(show_ship = false)
```

Comment:
Significant reduction! well done!! I'm going to be chewing on this to make I understand how it works. Good stuff. :)